rfdetr/datasets/yolo.py

```diff
@@ -0,0 +1,360 @@
+    )
+    contains_required_data_subdirs = all(
+        os.path.exists(os.path.join(dataset_dir, split_dir, data_subdir))
+        for split_dir in REQUIRED_SPLIT_DIRS
+        for data_subdir in REQUIRED_DATA_SUBDIRS
+    )
+    return contains_required_data_yaml and contains_required_split_dirs and contains_required_data_subdirs
+
+
+def build_yolo(image_set, args, resolution):
+    """Build YOLO dataset"""
+    root = Path(args.dataset_dir)
+    print(image_set)
+    PATHS = {
+        "train": (root / "train" / "images", root / "train" / "labels"),
+        "val": (root / "valid" / "images", root / "valid" / "labels"),
+        "test": (root / "test" / "images", root / "test" / "labels"),
+    }
+    
+    img_folder, labels_folder = PATHS[image_set.split("_")[0]]
+    data_yaml_path = root / "data.yaml"
+    
+    try:
+        square_resize_div_64 = args.square_resize_div_64
+    except:
+        square_resize_div_64 = False
+    
+    if square_resize_div_64:
+        dataset = YoloDetection(
+            img_folder, 
+            labels_folder,
+            data_yaml_path,
+            transforms=make_coco_transforms_square_div_64(
+                image_set, 
+                resolution, 
+                multi_scale=args.multi_scale, 
+                expanded_scales=args.expanded_scales
+            )
+        )
+    else:
+        dataset = YoloDetection(
+            img_folder, 
+            labels_folder,
+            data_yaml_path,
+            transforms=make_coco_transforms(
+                image_set, 
+                resolution, 
+                multi_scale=args.multi_scale, 
+                expanded_scales=args.expanded_scales
+            )
+        )
+    
+    return dataset 
+
+
+class YoloDetection(torch.utils.data.Dataset):
+    """Dataset for YOLO format annotations"""
+    def __init__(self, img_folder, labels_folder, data_yaml_path, transforms=None):
+        super(YoloDetection, self).__init__()
+        self.img_folder = img_folder
+        self.labels_folder = labels_folder
+        self.transforms = transforms
+        self.img_files = sorted(glob.glob(os.path.join(img_folder, '*.jpg')) + 
+                        glob.glob(os.path.join(img_folder, '*.jpeg')) + 
+                        glob.glob(os.path.join(img_folder, '*.png')))
+        
+        with open(data_yaml_path, 'r') as f:
+            data = yaml.safe_load(f)
+            self.class_names = data.get('names', [])
+            
+        print(f"Loaded {len(self.class_names)} classes from YOLO dataset: {self.class_names}")
+            
+        self.class_to_coco_id = {i: i for i in range(len(self.class_names))}
+        
+        self.ids = [i+1 for i in list(range(len(self.img_files)))]
+        
+        self._coco = None
+        
+    def __len__(self):
+        return len(self.img_files)
+
+    def _parse_yolo_annotations(self, lines: list[str], width: int, height: int) -> tuple[list, list]:
```

Comment:
please make it a stand alone public function; at some point we will want to add unit tests for it

also make it take `resolution_wh` as input argument